### ISSUE TITLE
model 中的 forward 方法的返回值问题,   PaddleNLP\examples\text_matching\ernie_matching\model.py

### ISSUE THREAD

**Issue by shguan2018** [PaddlePaddle/PaddleNLP#1325]
###########################################################
class PointwiseMatching(nn.Layer):
    def __init__(self, pretrained_model, dropout=None):
        super().__init__()
        self.ptm = pretrained_model
        self.dropout = nn.Dropout(dropout if dropout is not None else 0.1)

        # num_labels = 2 (similar or dissimilar)
        self.classifier = nn.Linear(self.ptm.config["hidden_size"], 2)

    def forward(self,
                input_ids,
                token_type_ids=None,
                position_ids=None,
                attention_mask=None):

        _, cls_embedding = self.ptm(input_ids, token_type_ids, position_ids,
                                    attention_mask)

        cls_embedding = self.dropout(cls_embedding)
        logits = self.classifier(cls_embedding)
        probs = F.softmax(logits)

        return probs
##########################################
forward 方法中，如果我想返回两个文本的相似度，应该怎么写呢?  我想在将来生成模型后，布置在 paddle serving 上, 预测时直接就调用这个 forward 方法,可以返回两个文本的相对度.(我不需要分类0,1，只需要相似度),   谢谢

**Comment by tianxin1860** [comment]
您好，预测是直接调用当前的 forward 方法会返回1个 2 维向量，分别表示 [不相似得分、相似度得分], 你可以取出向量的第2个元素作为2个文本的相似度即可。或者如果需要归一化，也可以在输出的 logits 基础上加上 softmax 进行归一化。
```
class PointwiseMatching(nn.Layer):
    def __init__(self, pretrained_model, dropout=None):
        super().__init__()
        self.ptm = pretrained_model
        self.dropout = nn.Dropout(dropout if dropout is not None else 0.1)

        # num_labels = 2 (similar or dissimilar)
        self.classifier = nn.Linear(self.ptm.config["hidden_size"], 2)

    def forward(self,
                input_ids,
                token_type_ids=None,
                position_ids=None,
                attention_mask=None):

        _, cls_embedding = self.ptm(input_ids, token_type_ids, position_ids,
                                    attention_mask)

        cls_embedding = self.dropout(cls_embedding)
        logits = self.classifier(cls_embedding)

        return logits
```

**Comment by shguan2018** [comment]
谢谢，我测试一下

**Comment by shguan2018** [comment]
curl -d '{"text": [["谁 有 狂三 这张 高清 的", "这张 高清图， 谁有"], ["英雄 联盟 什么 英雄 最好", "英雄 联盟 最好 英雄 是什么"]]}' 'http://127.0.0.1:8801/wenxin/inference'
{"content": [[0.9843046069145203, 0.01569538377225399], [0.005985298193991184, 0.9940146803855896]], "ret": 200}
按照你说的, 第一组文本相似度要是 取 0.015 不太合适
第二组文本相似度取  0.994 算合适吧,

**Comment by tianxin1860** [comment]
是的
